js/ui/grid_core/ui.grid_core.header_panel.js

```diff
@@ -187,6 +209,10 @@ const HeaderPanel = ColumnsView.inherit({
 export const headerPanelModule = {
     defaultOptions: function() {
         return {
+            toolbar: {
+                visible: true,
+                disabled: false
```

Comment:
I think that visible and disabled default values should not be assigned here. In this case visible will be calculated by toolbar.tems.length